packages/investor-foxy/src/api/api.ts

```diff
@@ -808,4 +809,69 @@ export class FoxyApi {
       releaseTime
     }
   }
+
+  async getRebaseHistory(input: BalanceInput) {
+    const { tokenContractAddress, userAddress } = input
+    this.verifyAddresses([tokenContractAddress])
+
+    const contract = new this.web3.eth.Contract(foxyAbi, tokenContractAddress)
+    const fromBlock = 14381454 // genesis rebase
+
+    const rebaseEvents = await (async () => {
+      try {
+        const events = (
+          await contract.getPastEvents('LogRebase', {
+            fromBlock,
+            toBlock: 'latest'
+          })
+        ).filter((rebase) => rebase.returnValues.rebase !== '0')
+        return events
+      } catch (e) {
+        throw new Error(`Failed to get rebase events ${e}`)
+      }
+    })()
+
+    const events: RebaseEvent[] = rebaseEvents.map((rebaseEvent) => {
+      const {
+        blockNumber,
+        returnValues: { epoch }
+      } = rebaseEvent
+      return {
+        blockNumber,
+        epoch
+      }
+    })
+
+    const rebaseChartData = await Promise.all(
+      events.map(async (event) => {
+        const balance = await (async () => {
+          try {
+            const postRebaseBalance = await contract.methods
+              .balanceOf(userAddress)
+              .call(null, event.blockNumber)
+            const preRebaseBalance = await contract.methods
+              .balanceOf(userAddress)
+              .call(null, event.blockNumber - 1)
+
+            return bnOrZero(postRebaseBalance).minus(preRebaseBalance).toString()
+          } catch (e) {
+            throw new Error(`Failed to get balance of address ${e}`)
+          }
+        })()
+
+        const timestamp = await (async () => {
+          try {
+            const block = await this.web3.eth.getBlock(event.blockNumber)
+            return block.timestamp
+          } catch (e) {
+            throw new Error(`Failed to get timestamp of block ${e}`)
+          }
+        })()
+
+        return { balance, timestamp }
+      })
+    )
+
+    return rebaseChartData
```

Comment:
do an `.every` on rebaseChartData to check for rejected promises, log an error, and return an empty array